statesync/stateprovider.go

```diff
@@ -179,12 +179,13 @@ func (s *lightClientStateProvider) State(ctx context.Context, height uint64) (sm
 		return sm.State{}, fmt.Errorf("unable to create RPC client: %w", err)
 	}
 	rpcclient := lightrpc.NewClient(primaryRPC, s.lc)
-	result, err := rpcclient.ConsensusParams(ctx, &nextLightBlock.Height)
+	result, err := rpcclient.ConsensusParams(ctx, &currentLightBlock.Height)
```

Comment:
Are you sure this is correct? Isn't `State.ConsensusParams` the consensus parameters that we will be using for the next block, not the ones that were used for the previous block? As such, we need to fetch them from `nextLightBlock`, since this is the next block that will be processed by the node, and thus the block that the consensus params will apply to?

Reply:
You can see this here, where the consensus params returned by block H are saved at block H+1 (which is the block they apply to):

https://github.com/tendermint/tendermint/blob/03a6fb277702bf6d7773d6519e23fed1f1933930/state/store.go#L193-L194

Reply:
Nevermind, reread the code and I think you're right since `currentLightBlock` is `height+1`. Good catch, off-by-one errors are the worst.

Reply:
Thanks for double checking. Yeah I found it difficult as well because there seems to be multiple areas where there could be a off-by-one error